pkg/recordset/recordset.go

```diff
@@ -225,6 +225,25 @@ func (m *Manager) createMissingTargetStacks(sourceStacks, targetStacks []string)
 }
 
 func (m *Manager) deleteOrphanTargetStacks(sourceStacks, targetStacks []string) error {
+	for _, target := range targetStacks {
+		found := false
+		targetClusterName := extractClusterName(target)
+		for _, source := range sourceStacks {
+			sourceClusterName := extractClusterName(source)
+			if sourceClusterName == targetClusterName {
+				found = true
+				break
+			}
+		}
+		if !found {
+			err := m.deleteTargetStack(target)
+			if err != nil {
+				m.logger.Log("level", "debug", "message", fmt.Sprintf("%q stack deleted", target))
```

Comment:
When the error is not `nil`, there is an error. So it should actually be returned to the caller. In case there is an error it can actually not mean that the stack got deleted, because an error implies deleting the stack failed.

Reply:
No, we want to remove as many stacks as possible, if we have an error on one of them we want to continue with the rest and retry the failed one in the next execution.